plutus-tx/src/Language/PlutusTx/Lift.hs

```diff
@@ -24,36 +27,47 @@ import qualified Language.PlutusIR.MkPir                       as PIR
 
 import qualified Language.PlutusCore                           as PLC
 import qualified Language.PlutusCore.Constant.Dynamic          as PLC
+import           Language.PlutusCore.Pretty                    (Pretty)
 import           Language.PlutusCore.Quote
 import qualified Language.PlutusCore.StdLib.Data.Function      as PLC
 import qualified Language.PlutusCore.StdLib.Meta.Data.Function as PLC
 
+import           Codec.Serialise
 import           Control.Exception
 import           Control.Monad.Except                          hiding (lift)
 import           Control.Monad.Reader                          hiding (lift)
 
 import           Data.Functor                                  (void)
 import           Data.Proxy
+import qualified Data.Typeable                                 as GHC
+
+type Throwable uni = (PLC.GShow uni, PLC.Closed uni, uni `PLC.Everywhere` Pretty, GHC.Typeable uni)
```

Comment:
I'm not sure about this. Let's say we need a `uni` that is `Throwable`, but also ``uni `PLC.Everywhere` Serialise``. If we have both these aliases

```haskell
type Serialisable uni = (PLC.Closed uni, uni `PLC.Everywhere` Serialise)
type Throwable uni = (PLC.GShow uni, PLC.Closed uni, uni `PLC.Everywhere` Pretty, GHC.Typeable uni)
```

and use them together

```haskell
(Serialisable uni, Throwable uni)
```

then `PLC.Closed uni` is used twice, which is kind of weird (also not sure what GHC thinks of that). Or should we omit it in one of the aliases or not use one of them and write e.g.

```haskell
(uni `PLC.Everywhere` Serialise, Throwable uni)
```

I tried using class aliases instead of type aliases (so that `PLC.Closed` is mentioned in the premise of each of the class aliases rather than twice in a type signature), but that broke something for whatever reason.

So this `Closed` thing is deeply irritating and I'm just not sure how to handle it in a sensible way, which is why I didn't do anything on the `language-plutus-core` side and deferred making type aliases to where I believe they're, em, disjoint.

I don't like the current situation either. But if we manage to get rid of `Closed`, that would make it easier to define disjoint constraints.

Reply:
I think we should try and see if GHC complains about the redundant TV.

Reply:
Just tried, doesn't seem to complain. Should I create those aliases? I'd still leave this for a future PR.

Reply:
Happy for it to be in a follow-up.